pkg/watch/filter.go

```diff
@@ -69,3 +73,37 @@ func (fw *filteredWatch) loop() {
 		}
 	}
 }
+
+// Recorder records all events that are sent from the watch until it is closed.
+type Recorder struct {
+	Interface
+
+	lock   sync.Mutex
+	events []Event
+}
+
+var _ Interface = &Recorder{}
+
+// NewRecorder wraps an Interface and records any changes sent across it.
+func NewRecorder(w Interface) *Recorder {
```

Comment:
I don't see references.  I'm not against having this API, but why do I have it here?